corehq/apps/cloudcare/static/cloudcare/js/form_entry/web_form_session.js

```diff
@@ -472,14 +474,24 @@ hqDefine("cloudcare/js/form_entry/web_form_session", [
         };
 
         self.changeLang = function (lang) {
-            this.serverRequest(
-                {
-                    'action': constants.CHANGE_LOCALE,
-                    'locale': lang,
-                },
-                function (resp) {
-                    $.publish('session.reconcile', [resp, lang]);
+            if (!self.session_id) {
+                // Handling for language change on case list and case search screens,
+                // which don't have associated session_ids.
+                hqRequire(["cloudcare/js/formplayer/menus/controller"], function (menusController) {
+                    var urlObject = utils.currentUrlToObject();
+                    urlObject.changeLang = lang;
+                    menusController.selectMenu(urlObject);
```

Comment:
yea I had thought about potentially refactoring out just the portion that I actually need (`applyListeners`) into its own function (and all the handler functions it routes to) but that seemed like much since this was scoped to just the case list/case search screens but you're right in that I think it'd be way better to make it cleaner so that any non-form screens can switch languages too